src/daft-core/src/schema.rs

```diff
@@ -18,7 +18,7 @@ use common_error::{DaftError, DaftResult};
 
 pub type SchemaRef = Arc<Schema>;
 
-#[derive(Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
```

Comment:
I'll take another look, IIRC, there was a reason I needed to add it, but i'll doublecheck!

Reply:
So this was because the new `project` method was taking an owned `self`, so I needed to copy the schema. I refactored to make `project take in `self: Arc<Self>` instead